### ISSUE TITLE
Regression from #179 in the core_renderer.php file

### ISSUE THREAD

**Issue by mkpelletier** [gjb2048/moodle-theme_essential#422]
I think there has been a regression with the user picture error.  That is, when a user has a message sitting in his inbox from another user who no longer exists in the database, the system throws an error.  I see the previous fix given here: https://github.com/gjb2048/moodle-theme_essential/commit/e21671df9327437b2daf668b1670da7d9b3c1a91 no longer exists in core_renderer.php.

adding this to line 394 of core_renderer.php resolves the issue.  For your reference, here is the debug output and backtrace:

Debug info: Argument 1 passed to user_picture::__construct() must be an instance of stdClass, boolean given, called in [dirroot]/theme/essential/renderers/core_renderer.php on line 394 and defined
Error code: codingerror
Stack trace:
	•	line 393 of /lib/setuplib.php: coding_exception thrown
	•	line 183 of /lib/outputcomponents.php: call to default_error_handler()
	•	line 394 of /theme/essential/renderers/core_renderer.php: call to user_picture->__construct()
	•	line 139 of /theme/essential/layout/includes/header.php: call to theme_essential_core_renderer->custom_menu_messages()
	•	line 27 of /theme/essential/layout/frontpage.php: call to require_once()
	•	line 877 of /lib/outputrenderers.php: call to include()
	•	line 807 of /lib/outputrenderers.php: call to core_renderer->render_page_layout()
	•	line 101 of /index.php: call to core_renderer->header()

**Comment by gjb2048** [comment]
What version of Essential please?

**Comment by gjb2048** [comment]
Was removed ages ago in: https://github.com/gjb2048/moodle-theme_essential/commit/b144f9d7d974268c7fca7f064da60e741eb4bb47#diff-492783807c33850a258efa189c59b5b2L352 for #225 so gob smacked it has not been seen before.

**Comment by mkpelletier** [comment]
Hello Gareth,

I believe I mentioned that it was 2.7.9.4.3 (Build: 2014111007).  Yes, it was removed, but in the latest version of the essential in Moodle 2.7.3, I got this in an isolated user account.  I was able to replicate by creating two user accounts, sending a message between the accounts, and then deleting one of these accounts from the database.  Note, that one would actually need to delete the user record from mdl_user, since a user deleted from the GUI remains in the table.  I realise this means that the error should not crop up, but if one comes from a previous version of Moodle where deleted users are not maintained in the db, one can encounter this error. 

I hope that helps.

Thanks, 

Mathieu
